### ISSUE TITLE
Special diagnostics & logging

### ISSUE THREAD

**Issue by laheller** [dsccommunity/xDscDiagnostics#58]
Hi!

As part of DSC apply/verify monitoring process I would like to know EXACTLY, which DSC resource INSTANCE (given by a unique name) is currently processed. So far the ONLY way I found to achieve this, is to enable/read the **Microsoft-Windows-Desired State Configuration/Analytic** log with EventID equals to **4162** which fulfills the above requirement.

Example log, where we can see which DSC resource instance is beeing processed:
![image](https://github.com/dsccommunity/xDscDiagnostics/assets/5788998/d390fb46-931c-4074-93d5-875452579c24)

Via PowerShell it's easy to read:
`Get-WinEvent -FilterHashtable @{ LogName = 'Microsoft-Windows-DSC/Analytic'; Id = 4162 } -Oldest`

I also figured out, how to detect the current working DSC job GUID at runtime:
`$JobGUID = [Microsoft.PowerShell.DesiredStateConfiguration.Internal.S_DscCoreR]::JobGuidString`

But the problem is that I don't want to use event logs at all for diagnostics.
My idea is to do this kind of logging directly at DSC apply/verify runtime form inside the custom DSC resource.
Example (one instance of the DSC resource in a DSC configuration):

```
MyCustomResource "MyInstanceName" {
	Property1 = "value1";
	Property2 = 5;
	Property3 = @("value3","value4")
}

```
Then from the **Test( )** method of **MyCustomResource** I would like to get somehow the instance name: _“[MyCustomResource]MyInstanceName”_
Something like:

```
[bool] Test() {
	$InstanceID = ... # the value of $InstanceID should be  "[MyCustomResource]MyInstanceName"
	return $result
}

```
Basically the question is, how to **programatically** get the instance name (in the above example the string
_"[MyCustomResource]MyInstanceName"_) from the custom resource's Set( ) or Test( ) method?

What I tried so far:
Using `Enable-DscDebug` to debug my custom dsc resource and when stopped at whatever breakpoint, I tried to check all possible PowerShell (automatic) variables. Unfortunately none of the holds the value I am looking for.

BR,

Ladislav

**Comment by johlju** [comment]
Not sure that is possible since LCM has the data and LCM does not pass that to the actual resources that are ran. You can possible get the resource name but not what the instance was named in the DSC configuration. You can try asking in the Slack/Discord #DSC channel https://dsccommunity.org/community/contact/

**Comment by laheller** [comment]
@johlju 
Finally I found the solution to my problem. It is described in [this](https://dsccommunity.org/blog/how-to-use-dsc-logging/) blog post at the section _Reading the active log file_.
